2020-05-05 Gurbir Arora <gurbir@lnxrov09.seas.ucla.edu> 

 * timespec.h (timespec_cmp): Avoids integer overflow by return 1 if A>B, -1 if A<B, and 0 if they are equal.

--- lib/timespec.h	2020-05-08 02:00:34.563215000 -0700
+++ lib/timespec.h	2020-05-08 02:00:44.466975000 -0700
@@ -44,8 +44,12 @@
    Assume the nanosecond components are in range, or close to it.  */
 static inline int
 timespec_cmp (struct timespec a, struct timespec b){
-	int diff = a.tv_sec - b.tv_sec;
-	return diff ? diff : a.tv_nsec - b.tv_nsec;
+	if(a.tv_sec > b.tv_sec){
+		return 1; 
+	} else if (a.tv_sec < b.tv_sec) {
+		return -1; 
+	}  
+	return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
